### PR TITLE
[Core] Convert point output process

### LINKED ISSUE #7877: [Core] PointOutputProcess is not an output process (no IsOutputStep nor PrintOutput)
**Description**
An update of the visualization_variables of elements, which are cut, of the visualization_model_part does not seem to happen every time step. E.g. if a PointOutputProcess (or one of the derived classes) creates output every time step for a cut element, the values in the output file of the PointOutputProcess won't change for a certain number of time steps (often until a GiD or VTK Output of the visualization model part happens). The output of works for the standard FluidModelPart.fluid_computational_model_part.

**Scope**
- EmbeddedSkinVisualizationProcess

**To Reproduce**
- Use `embedded_symbolic_navier_stokes_discontinuous` as element_type
- Add `ApplyEmbeddedSkinVisualizationProcess` in your ProjectParameters to have a visualization_model_part,( e.g. "visualization_variables": ["VELOCITY","PRESSURE"])
- Add a `PointOutputProcess` using the visualization_model_part for a point in an element, which is cut (e.g. "output_variables": [ "PRESSURE"])
- Add a `GiDOutputProcess`, which output_interval is bigger than 1 (e.g. "nodal_results"       : ["VELOCITY","PRESSURE"])
- Run a couple of time steps (more that output_interval of your GiDOutputProcess)
example see [visOutput.zip](https://github.com/KratosMultiphysics/Kratos/files/5599021/visOutput.zip) --> first output point of the LineOutputProcess is a cut element, the remaining points not

**Expected behavior**
The files created by a PointOutputProcess writes updated results for every time step, including elements, which are cut.

**Environment**
- OS: Ubuntu 20.04.1
- Branch: master
- Python 3.7.4

### PR DESCRIPTION
**Pull request by frawahl**
**📝 Description**
This PR converts `PointOutputProcess` to be an actual `KratosMultiphysics.OutputProcess` and adapts affiliated output processes as well as their tests.

**🆕 Changelog**
- Converted `PointOutputProcess` and `MultiplePointsOutputProcess` to be derived from `OutputProcess`
- Adapted `LineOutputProcess`, `GiDOutputProcess`, `GiDDamOutputProcess`
- Adapted `test_processes.py` and its function `SolutionLoopPointOutputProcesses` to accommodate `OutputProcess`es
- Fixed #7877, 
made possible by #9529

### CODE REVIEW

File: kratos/python_scripts/multiple_points_output_process.py [hunk 1]
```diff
@@ -83,6 +83,13 @@ def ExecuteBeforeOutputStep(self):
         for proc in self.point_output_processes:
             proc.ExecuteBeforeOutputStep()
 
+    def IsOutputStep(self):
+        return self.point_output_processes[0].IsOutputStep()
```
Comment (rubenzorrilla) [inline kratos/python_scripts/multiple_points_output_process.py:88]:
```suggestion
        # We check only the first process as the interval is the same in all of them
        return self.point_output_processes[0].IsOutputStep()
```
Just an annotation for the future.

File: kratos/python_scripts/line_output_process.py [hunk 2]
```diff
@@ -5,22 +5,13 @@
 from KratosMultiphysics.multiple_points_output_process import MultiplePointsOutputProcess
 
 def Factory(settings, Model):
-    if(type(settings) != KratosMultiphysics.Parameters):
+    if not isinstance(settings, KratosMultiphysics.Parameters):
         raise Exception("expected input shall be a Parameters object, encapsulating a json string")
-    return LineOutputProcess(Model, settings["Parameters"])
-
-class LineOutputProcess(KratosMultiphysics.Process):
```
Comment (rubenzorrilla) [inline kratos/python_scripts/line_output_process.py:12]: Why is the class removed?

Comment (rubenzorrilla) [reply kratos/python_scripts/line_output_process.py:12]: My proposal would be to derive this `LineOutputProcess` from the  `MultiplePointsOutputProcess` so you can get rid of all the calls below (as you already did).

Then we keep the `Factory` function as it was before and we only implement the constructor as well as the `GetDefaultParameters`. Once the input parameters have been validated, we tweak them so the call of the base process constructor doesn't crash.

**Review by rubenzorrilla — COMMENTED** [review]
I'm very positive about these changes. However, there are a few comments with regard to the `LineOutputProcess`.

**Comment by rubenzorrilla** [comment]
Just mentioning #9910 as it might be relevant in here.

File: kratos/python_scripts/line_output_process.py (hunk 2 above)
Comment (frawahl) [reply kratos/python_scripts/line_output_process.py:12]: Sure, I can derive it from `MultiplePointsOutputProcess`.
I'm not sure about the advantages/ disadvantages ...
I changed the `LineOutputProcess` to what @philbucher suggested for the `CSVPointsOutputProcess`, which actually is the reason I didn't have to change anything now for the `CSVPointsOutputProcess`.

Comment (rubenzorrilla) [reply kratos/python_scripts/line_output_process.py:12]: IMO it is more object oriented solution as Kratos is, thus making possible to create an instance of the object by directly calling the constructor of the class. On top of this, setting `LineOutputProcess` in the json settings but getting a `MultiplePointsOutputProcess` instance in your process list might be misleading...

Comment (rubenzorrilla) [reply kratos/python_scripts/line_output_process.py:12]: OFC this is a personal opinion. It is clear that current solution works.

Comment (frawahl) [reply kratos/python_scripts/line_output_process.py:12]: Should I change CSVPointsOutputProcess then as well? Doesn't it make sense to be consistent?

File: kratos/python_scripts/gid_output_process.py [hunk 3]
```diff
@@ -275,6 +275,8 @@ def IsOutputStep(self):
     def PrintOutput(self):
         if self.point_output_process is not None:
             self.point_output_process.ExecuteBeforeOutputStep()
+            if self.point_output_process.IsOutputStep():
```
Comment (matekelemen) [inline kratos/python_scripts/gid_output_process.py:278]: As far as I know, the idea behind```OutputProcess::PrintOutput``` is to write the output without checking ```OutputProcess::IsOutputStep```.

```PythonSolver``` already [performs the check](https://github.com/KratosMultiphysics/Kratos/blob/c1fb8e578ba79317b88f3c519cb5e055780f559d/kratos/python_scripts/analysis_stage.py#L157) directly in the solution loop once, and then calls ```PrintOutput``` if successful. I don't think it's good practice, but @sunethwarna tells me that some people rely on this specific behaviour.

Comment (frawahl) [reply kratos/python_scripts/gid_output_process.py:278]: Yes, you are right, that was a mistake.

Same goes for gid_dam_output_process of course.

File: kratos/python_scripts/line_output_process.py (hunk 2 above)
Comment (rubenzorrilla) [reply kratos/python_scripts/line_output_process.py:12]: I'd leave it as it is, at least in this PR.

**Comment by mpentek** [comment]
Great to see progress on this!

File: kratos/python_scripts/gid_output_process.py (hunk 3 above)
Comment (frawahl) [reply kratos/python_scripts/gid_output_process.py:278]: I just noticed, there was reasoning behind this .. 
you are right that the `PrintOutput` of `GiDOutputProcess` will only be called if `IsOutputStep` returns true. But analysis_stage won't know about `PointOutputProcess`es inside `GiDOutputProcess` and their output interval will either be checked inside the `PrintOutput` of `GiDOutputProcess` or not at all and the interval could be different, right? (I don't know what these point outputs inside the GiD output are used for)

**Review by rubenzorrilla — APPROVED** [review]

**Pull request merged**